src/platform/common/application/notebook.ts

```diff
@@ -98,4 +108,18 @@ export class VSCodeNotebook implements IVSCodeNotebook {
     ): NotebookController {
         return notebooks.createNotebookController(id, viewType, label, handler, rendererScripts);
     }
+
+    @testOnlyMethod()
+    public async closeActiveNotebooks() {
```

Comment:
Tests weren't passing because the 'revertAndCloseActiveEditor' doesn't fire the OnDidCloseNotebook event. This forces tests to fire the close event. 

Without the close event, the old coordinator from the old notebook document is still attached.